x/supply/internal/types/permissions.go

```diff
@@ -10,11 +10,14 @@ const (
 )
 
 // validate the input permissions
-func validatePermissions(permission string) error {
-	switch permission {
-	case Basic, Minter, Burner:
-		return nil
-	default:
-		return fmt.Errorf("invalid module permission %s", permission)
+func validatePermissions(permissions []string) error {
```

Comment:
What if the permissions registered on initialization are not part of the basic set?

Reply:
Maybe I should move this func to `Keeper` and require a array of allowedPermissions to be passed into `NewKeeper` and then just check if the permission is in that array. This would allow permissions to be registered upon creation of the supply keeper

Reply:
On second thought, the `permAddrs` map already acts as allowedPermissions. Can just check that the permissions passed in also exist in that map

Reply:
Sounds good @colin-axner; we don't want to restrict there permission set here (e.g. `Basic, Minter, Burner, Staking`).